### PR TITLE
PM4 Update

### PR DESCRIPTION
**Pull request by ItsToxicGG**

### CODE REVIEW

**Comment by ItsToxicGG** [comment]
NEEDS TO BE TESTED

File: src/aliuly/itemcasepe/Main.php [hunk 1]
```diff
@@ -119,15 +119,15 @@ private function cmdAdd(CommandSender $c): bool {
     }
 
     private function cmdRespawn(): bool {
-        foreach ($this->getServer()->getLevels() as $lv) {
-            $world = $lv->getName();
+        foreach ($this->getServer()->getWorldManager()->getWorlds() as $lv) {
+            $world = $this->getServer()->getWorldManager()->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:122]: are you sure this is correct?

File: src/aliuly/itemcasepe/Main.php [hunk 2]
```diff
@@ -119,15 +119,15 @@ private function cmdAdd(CommandSender $c): bool {
...
             $players = $lv->getPlayers();
             foreach (array_keys($this->cases[$world]) as $cid) {
                 $this->rmItemCase($lv, $cid, $players);
             }
         }
-        foreach ($this->getServer()->getLevels() as $lv) {
-            $world = $lv->getName();
+        foreach ($this->getServer()->getWorldManager()->getWorlds() as $lv) {
+            $world = $this->getServer()->getWorldManager()->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:129]: same here

File: src/aliuly/itemcasepe/Main.php [hunk 3]
```diff
@@ -140,23 +140,23 @@ private function cmdRespawn(): bool {
     // Place/Remove ItemCases
     //
     ////////////////////////////////////////////////////////////////////////
-    private function rmItemCase(Level $level, $cid, array $players) {
+    private function rmItemCase(World $level, $cid, array $players) {
         //echo __METHOD__.",".__LINE__."\n";
-        $world = $level->getName();
+        $world = $level->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:144]: why?

File: src/aliuly/itemcasepe/Main.php [hunk 4]
```diff
@@ -233,8 +233,8 @@ private function saveCfg(Level $lv) {
         file_put_contents($f, $dat);
     }
 
-    private function loadCfg(Level $lv) {
-        $world = $lv->getName();
+    private function loadCfg(World $lv) {
+        $world = $lv->getWorldManager()->getWorldByName;
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:236]: ... please check

File: src/aliuly/itemcasepe/Main.php [hunk 5]
```diff
@@ -219,8 +219,8 @@ public function addItemCase(Level $level, $cid, $idmeta, $count): bool {
         return true;
     }
 
-    private function saveCfg(Level $lv) {
-        $world = $lv->getName();
+    private function saveCfg(World $lv) {
+        $world = $lv->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:222]: same here, should be folder name instead?

File: src/aliuly/itemcasepe/Main.php [hunk 6]
```diff
@@ -198,17 +198,17 @@ public function spawnLevelItemCases(Level $level) {
...
         foreach (array_keys($this->cases[$world]) as $cid) {
             $this->rmItemCase($level, $cid, [$pl]);
         }
     }
 
-    public function addItemCase(Level $level, $cid, $idmeta, $count): bool {
+    public function addItemCase(World $level, $cid, $idmeta, $count): bool {
         //echo __METHOD__.",".__LINE__."\n";
-        $world = $level->getName();
+        $world = $level->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:210]: same gere

File: src/aliuly/itemcasepe/Main.php [hunk 7]
```diff
@@ -198,17 +198,17 @@ public function spawnLevelItemCases(Level $level) {
         }
     }
 
-    public function despawnPlayerCases(Player $pl, Level $level) {
+    public function despawnPlayerCases(Player $pl, World $level) {
         $world = $level->getName();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:201]: did you forget to change this

File: src/aliuly/itemcasepe/Main.php [hunk 8]
```diff
@@ -165,11 +165,11 @@ private function sndItemCase(Level $level, $cid, array $players) {
         $item = Item::fromString($this->cases[$world][$cid]["item"]);
         $item->setCount($this->cases[$world][$cid]["count"]);
         $pk = new AddItemActorPacket();
-        $pk->entityRuntimeId = $this->cases[$world][$cid]["eid"];
+        $pk->actorRuntimeId = $this->cases[$world][$cid]["eid"];
         $pk->item = ItemStackWrapper::legacy($item);
         $pk->position = new Vector3($pos[0] + 0.5, (float)$pos[1] + 0.25, $pos[2] + 0.5);
         $pk->motion = new Vector3(0, 0, 0);
-        $pk->metadata = [Entity::DATA_FLAGS => [Entity::DATA_TYPE_LONG, 1 << Entity::DATA_FLAG_IMMOBILE]];
+        $pk->metadata = [\pocketmine\network\mcpe\protocol\types\entity\EntityMetadataProperties::FLAGS => [\pocketmine\network\mcpe\protocol\types\entity\EntityMetadataTypes::LONG, 1 << \pocketmine\network\mcpe\protocol\types\entity\EntityMetadataFlags::IMMOBILE]];
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:171]: why don't you import the class?

File: src/aliuly/itemcasepe/Main.php [hunk 9]
```diff
@@ -140,23 +140,23 @@ private function cmdRespawn(): bool {
...
         foreach ($players as $pl) {
             $pl->directDataPacket($pk);
         }
     }
 
-    private function sndItemCase(Level $level, $cid, array $players) {
+    private function sndItemCase(World $level, $cid, array $players) {
         //echo __METHOD__.",".__LINE__."\n";
-        $world = $level->getName();
+        $world = $level->getWorld();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:158]: same here

File: src/aliuly/itemcasepe/Main.php [hunk 10]
```diff
@@ -253,24 +253,24 @@ private function loadCfg(Level $lv) {
...
-    public function onLevelLoad(LevelLoadEvent $e) {
-        $this->loadCfg($e->getLevel());
+    public function onLevelLoad(WorldLoadEvent $e) {
+        $this->loadCfg($e->getWorld());
     }
 
-    public function onLevelUnload(LevelUnloadEvent $e) {
-        $world = $e->getLevel()->getName();
+    public function onLevelUnload(WorldUnloadEvent $e) {
+        $world = $e->getWorld()->getName();
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:260]: change to folder name

File: src/aliuly/itemcasepe/Main.php [hunk 11]
```diff
@@ -281,7 +281,7 @@ public function onSendPacket(DataPacketSendEvent $ev) {
         }
         // Re-spawn as chunks get sent...
         $pl = $ev->getPlayer();
-        $level = $pl->getLevel();
+        $level = $pl->getWorld();
         if (!isset($this->cases[$level->getName()])) return;
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:284]: change to folder name

File: src/aliuly/itemcasepe/Main.php [hunk 12]
```diff
@@ -312,8 +312,8 @@ public function onPlayerInteract(PlayerInteractEvent $ev) {
         if (!isset($this->touches[$pl->getName()])) return;
         $bl = $ev->getBlock();
         if ($this->classic) {
-            if ($bl->getID() != Block::GLASS) {
-                if ($bl->getID() == Block::STONE_SLAB) {
+            if ($bl->getID() != \pocketmine\block\BlockLegacyIds::GLASS) {
+                if ($bl->getID() == \pocketmine\block\BlockLegacyIds::STONE_SLAB) {
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:315]: import the class

File: src/aliuly/itemcasepe/Main.php [hunk 13]
```diff
@@ -322,19 +322,19 @@ public function onPlayerInteract(PlayerInteractEvent $ev) {
...
         } else {
-            if ($bl->getID() != Block::GLASS) {
+            if ($bl->getID() != \pocketmine\block\BlockLegacyIds::GLASS) {
                 $bl = $bl->getSide(Vector3::SIDE_UP);
             }
         }
         $cid = implode(":", [$bl->getX(), $bl->getY(), $bl->getZ()]);
         $item = $pl->getInventory()->getItemInHand();
-        if ($item->getId() === Item::AIR) {
+        if ($item->getId() === \pocketmine\item\ItemIds::AIR) {
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php:330]: import the class

File: src/aliuly/itemcasepe/Main.php [hunk 14]
```diff
@@ -352,21 +352,21 @@ public function onPlayerInteract(PlayerInteractEvent $ev) {
...
     }
 
     public function onBlockBreak(BlockBreakEvent $ev) {
         $pl = $ev->getPlayer();
         $bl = $ev->getBlock();
-        $lv = $bl->getLevel();
-        $yoff = $bl->getId() != Block::GLASS ? 1 : 0;
+        $lv = $bl->getWorld();
+        $yoff = $bl->getId() != \pocketmine\block\BlockLegacyIds::GLASS ? 1 : 0;
         $cid = implode(":", [$bl->getX(), $bl->getY() + $yoff, $bl->getZ()]);
```
Comment (SOF3) [inline src/aliuly/itemcasepe/Main.php]: import

**Review by SOF3 — CHANGES_REQUESTED** [review]

**Comment by SOF3** [comment]
you're just adding random files at this point. seeing no reply to the aforementioned issues I'm closing this PR.

**Comment by ItsToxicGG** [comment]
True im out of idea I'm gonna fix it with 0 lints

**Comment by SOF3** [comment]
Please stick to the topic. If your PR is about updating for PM4, please don't include changes that are not related to updating for PM4.

**Comment by SOF3** [comment]
What does `WorldC.php` even do?

**Pull request closed without merging**